analyses/bangladesh/trigger_development/bgd-glofas-new-model.md

```diff
@@ -0,0 +1,404 @@
+```python
+from importlib import reload
+from pathlib import Path
+import os
+from datetime import datetime
+
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+import pandas as pd
+import numpy as np
+from scipy.interpolate import interp1d
+from scipy.stats import pearsonr
+
+import read_in_data as rd
+reload(rd)
+
+mpl.rcParams['figure.dpi'] = 200
+
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"])
+SKILL_DIR = DATA_DIR / 'exploration/bangladesh/GLOFAS_Data'
+SKILL_FILE = 'forecast_skill.csv'
+LEADTIMES_V2 = [5, 10, 15, 20, 25, 30]
+MAIN_VERSION = 3
+```
+
+### Create GloFAS objects
+
+```python
+da_glofas_reanalysis = {
+    2: rd.get_glofas_reanalysis(version=2),
+    3: rd.get_glofas_reanalysis()
+}
+
+da_glofas_forecast = {
+    2: rd.get_glofas_forecast(version=2, leadtimes=LEADTIMES_V2),
+    3: rd.get_glofas_forecast(),
+
+}
+
+da_glofas_reforecast = {
+    2: rd.get_glofas_reforecast(version=2, interp=False, leadtimes=LEADTIMES_V2),
+    3: rd.get_glofas_reforecast(interp=False)
+}
+
+da_glofas_reforecast_interp = {
+    2: rd.get_glofas_reforecast(version=2, leadtimes=LEADTIMES_V2),
+    3: rd.get_glofas_reforecast()
+}
+```
+
+### Calculate return periods
+
+```python
+def get_return_period_function(observations):
+    df_rp = (observations.to_dataframe()[[rd.STATION]]
+                 .rename(columns={rd.STATION: 'discharge'})
+                 .resample(rule='A', kind='period')
+                 .max() 
+                 .sort_values(by='discharge', ascending=False)
+                )
+    df_rp["year"] = df_rp.index.year
+     
+    n = len(df_rp)
+    df_rp['rank'] = np.arange(n) + 1
+    df_rp['exceedance_probability'] = df_rp['rank'] / (n+1)
+    df_rp['rp'] = 1 / df_rp['exceedance_probability']
+    return interp1d(df_rp['rp'], df_rp['discharge'])
+
+rp_dict = {}
+for version in [2,3]:
+    f_rp = get_return_period_function(da_glofas_reanalysis[version])
+    print(f'Version {version}')
+    for year in [1.5, 2, 3, 4, 5, 10, 20]:
+        val = 5000*np.round(f_rp(year) / 5000)
+        print(year, val)
+        if version == MAIN_VERSION:
+            rp_dict[year] = val
+# Make 1 in 5 year always 100,000
```

Comment:
How come these values are being manually set?

Reply:
For the new model, the 1 in 5 y RP is around 97000, so rounds down to 95000. But since it's so close to the model version 2 1 in 5 year value of 98000, we just round them both up to 100,000. I've added an explantion.